### ISSUE TITLE
Button with Image on iOS and MacOS always missalign

### ISSUE THREAD

**Issue by arkadym** [dotnet/maui#8528]
### Description

Not possible to use Image with Button control it always missalign (see screenshots). Image always align to edge w/o any padding. 
Using padding will not help because there always extra space in between image and text

![Screenshot 2022-07-05 at 12 13 15](https://user-images.githubusercontent.com/7422012/177255677-153f7296-ebec-4e94-b88a-46e131c5277d.png)
![Screenshot 2022-07-05 at 12 13 07](https://user-images.githubusercontent.com/7422012/177255680-832dbd75-5df1-4e14-bdb3-6ca406e1dd98.png)
![Screenshot 2022-07-05 at 12 12 50](https://user-images.githubusercontent.com/7422012/177255683-c1484d68-8994-4062-baba-0cf3c6243c8b.png)
![Screenshot 2022-07-05 at 12 15 20](https://user-images.githubusercontent.com/7422012/177255667-add5eef1-130d-4803-a618-be8ac927ec4a.png)

### Steps to Reproduce

1. Create NET MAUI App
2. Add any image to Click Me button (have to run on iOS or MacOS))
3. Run app

### Version with bug

6.0.400 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, macOS

### Affected platform versions

iOS 15, MacOS 12.3

### Did you find any workaround?

No workaround found yet

**Comment by jsuarezruiz** [comment]
The space between the icon and the text is larger than it was in Xamarin.Forms.
[Issue8528.zip](https://github.com/dotnet/maui/files/9044831/Issue8528.zip)
<img width="549" alt="Captura de Pantalla 2022-07-05 a las 10 45 20" src="https://user-images.githubusercontent.com/6755973/177288544-e6006f8d-340b-4fbf-b8f0-d26bfbac5223.png">

**Comment by davidortinau** [comment]
Yes, I'm seeing this also on Catalyst. No matter how wide I make the button or what padding is set, the image is at the edge.

**Comment by davidortinau** [comment]
My workaround is to compose an Image and a Label

```
<Border Stroke="{StaticResource SecondaryBrush}" StrokeShape="{RoundRectangle CornerRadius=8}"
                                StrokeThickness="1" Background="Transparent"
                                Grid.Row="0" HorizontalOptions="End" VerticalOptions="Start"
                                SemanticProperties.Description="Button that opens a filter dialog"
                                >
                            <HorizontalStackLayout Margin="14,12" Spacing="12">
                                <Image Source="filter.png" VerticalOptions="Center" />
                                <Label Text="Filter" VerticalOptions="Center"/>
                            </HorizontalStackLayout>

                            <Border.GestureRecognizers>
                                <TapGestureRecognizer Tapped="TapGestureRecognizer_Tapped"/>
                            </Border.GestureRecognizers>
                        </Border>
```

So instead of this:

<img width="173" alt="Screen Shot 2022-07-20 at 4 34 06 PM" src="https://user-images.githubusercontent.com/41873/180087121-b1c97299-e452-4082-ab74-b8863f552723.png">


I get this:

<img width="130" alt="Screen Shot 2022-07-20 at 4 37 34 PM" src="https://user-images.githubusercontent.com/41873/180087146-e2137c3b-e69b-4c91-b5db-9fa8f7990b59.png">

**Comment by arkadym** [comment]
Yes, is good idea, but i prefer keep it original, e.g. button has hover effect, click effect. So I've got another workaround - i realized negative padding works fine for image location:
```
ContentLayout="{OnPlatform Default='Top,2', iOS='Top,-10', MacCatalyst='Top,-10'}"
Padding="{OnPlatform Default='0,3', iOS='0,15,0,0', MacCatalyst='0,15,0,0'}" 
```

**Comment by AbstractionsAs** [comment]
Is there any progress on this?

**Comment by hartez** [comment]
This should have been closed when #10309 was merged.
